### PR TITLE
Introducing Modal component

### PR DESCRIPTION
**Pull request by wmurray**
This introduces a new Modal component to anchor-ui. It is a simplified version of the Modal components found in Wharf/Buoy, preventing users from having to compose multiple pieces of the modal in order to create the modal in its entirety. 

Additionally, it introduces a compatibility component, LegacyModal, which can be used as a drop-in replacement for the current Modal component.

<details><summary>Screenshot</summary>

Screenshot 2023-03-15 at 5 16 48 PM
Screenshot 2023-03-15 at 5 16 34 PM
</details>

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204058824672272

### CODE REVIEW

File: packages/components/modal/__specs__/modal.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            </Button>
+          </>
+        }
+        onRequestClose={jest.fn()}
+        testId="test-modal"
+        title="Test Modal"
+      />
+    );
+
+    expect(modal.asFragment()).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (wmurray) [reply packages/components/modal/__specs__/modal.spec.tsx:27]: Got the spec running - forgot a few config files. It's choking on `ReactModal.setAppElement("#root");` in the component (I have a `<Box id="root"...>` wrapping it in the spec).

**Comment by wmurray** [comment]
Converting to draft - need to expand the functionality by including a LegacyModal for a more plug-and-play migration path.

File: jest.config.ts [hunk 2]
```diff
@@ -25,7 +26,9 @@ const reactMarkdownEsModules = [
   "zwitch",
 ];
 
-const esModules = [...reactMarkdownEsModules].join("|");
+const esModules = [...reactMarkdownEsModules, "@buoysoftware/anchor-.+"].join(
```
Comment (wmurray) [inline jest.config.ts:29]: In order to get the modal spec to pass, we had to add anchor-* to the transformIgnorePatterns. We did this at the root level, but it stands to reason we could update this to be for individual packages or components instead. WDUT?

File: jest.config.ts [hunk 3]
```diff
@@ -13,6 +13,7 @@ const reactMarkdownEsModules = [
   "micromark",
   "property-information",
   "react-markdown",
+  "react-modal",
```
Comment (acostanzo) [inline jest.config.ts:16]: I think this belongs on line 29 similar to "@buoysoftware/anchor-.+". This array sees to cover the peer dependencies of react markdown

File: packages/components/legacy_modal/jest.setup.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import ReactModal from "react-modal";
+import "@testing-library/jest-dom";
+import "jest-styled-components";
+
+jest.spyOn(ReactModal, "setAppElement").mockImplementation(() => undefined);
```
Comment (acostanzo) [inline packages/components/legacy_modal/jest.setup.ts:5]: If we are mocking the implementation of setAppElement entirely do we still need to call it from the test and wrap the fake parent element around the modal?

File: packages/components/modal/jest.setup.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import ReactModal from "react-modal";
+import "@testing-library/jest-dom";
+import "jest-styled-components";
+
+jest.spyOn(ReactModal, "setAppElement").mockImplementation(() => undefined);
```
Comment (acostanzo) [inline packages/components/modal/jest.setup.ts:5]: Same question on calling setAppElement

File: packages/components/modal/src/modal.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+          backgroundColor: contentBackground,
+          border: "none",
+          borderRadius: contentRadius,
+          bottom: undefined,
+          left: undefined,
+          marginTop: space.modal.gutterT,
+          padding: 0,
+          position: undefined,
+          right: undefined,
+          top: undefined,
```
Comment (acostanzo) [inline packages/components/modal/src/modal.tsx:58]: Are these styles a sort of override of some default? If not can they be dropped?

**Review by acostanzo — COMMENTED** [review]
Looking good! Just a few questions

File: jest.config.ts (hunk 3 above)
Comment (wmurray) [reply jest.config.ts:16]: Makes sense!

File: packages/components/legacy_modal/jest.setup.ts (hunk 4 above)
Comment (wmurray) [reply packages/components/legacy_modal/jest.setup.ts:5]: Removing the parent element/call to setAppElement in `jest.setup` still allows the modal to render correctly. I think we can drop the wrapping element and call to `setAppElement`.

File: packages/components/modal/src/modal.tsx (hunk 6 above)
Comment (wmurray) [reply packages/components/modal/src/modal.tsx:58]: Gonna go ahead and say yeah, they're overrides 😂 

Screenshot 2023-03-15 at 5 01 11 PM

Comment (wmurray) [reply packages/components/modal/src/modal.tsx:58]: Good call, though - I missed some spacing updates that were needed to bring this in line with the new theme. Those have also been updated!

**Review by acostanzo — APPROVED** [review]

**Pull request merged**